### PR TITLE
First pass reuse revisions

### PR DESCRIPTION
**Pull request by Maxusmusti**

### CODE REVIEW

File: controllers/appwrapper_controller.go [hunk 1]
```diff
@@ -488,44 +486,54 @@ func checkIfExactMatchExists(aw *arbv1.AppWrapper) bool {
...
 }
 
 func onDelete(obj interface{}) {
 	aw, ok := obj.(*arbv1.AppWrapper)
 	if ok {
 		klog.Infof("Appwrapper deleted scale-down machineset: %s ", aw.Name)
 		if reuse {
-			if checkIfExactMatchExists(aw) {
-				scaleDown(aw)
+			match := findExactMatch(aw)
```
Comment (asm582) [inline controllers/appwrapper_controller.go]: Nit: change name to matchAppwrapper

File: controllers/appwrapper_controller.go [hunk 2]
```diff
@@ -488,44 +486,54 @@ func checkIfExactMatchExists(aw *arbv1.AppWrapper) bool {
...
 			}
 		}
 	}
-	return shouldScaleDown
+	return match
+
+}
 
+// add logic to swap out labels with new appwrapper label
+func swapLabels(oldAw *arbv1.AppWrapper, newAw *arbv1.AppWrapper) {
```
Comment (asm582) [inline controllers/appwrapper_controller.go]: Nit: change function name to swapNodeLabels

File: controllers/appwrapper_controller.go [hunk 3]
```diff
@@ -488,44 +486,54 @@ func checkIfExactMatchExists(aw *arbv1.AppWrapper) bool {
...
+				addLabelToNode(newAw, nodeName)
+			}
+		}
+	}
 }
 
 func onDelete(obj interface{}) {
 	aw, ok := obj.(*arbv1.AppWrapper)
 	if ok {
 		klog.Infof("Appwrapper deleted scale-down machineset: %s ", aw.Name)
```
Comment (asm582) [inline controllers/appwrapper_controller.go]: reposition the logger

**Review by asm582 — APPROVED** [review]
Thanks, Looks good, please address Nits, Have we tested it post-code changes?

File: controllers/appwrapper_controller.go (hunk 1 above)
Comment (Maxusmusti) [reply controllers/appwrapper_controller.go]: Do you mean change `match` or `findExactMatch`

Comment (asm582) [reply controllers/appwrapper_controller.go]: match := findExactMatch(aw) --> matchedAppwrapper :=findExactMatch(aw)

**Review by asm582 — APPROVED** [review]
LGTM

File: controllers/appwrapper_controller.go [hunk 4]
```diff
@@ -208,7 +208,7 @@ func canScale(demandPerInstanceType map[string]int) bool {
 		totalNodesAddedByMachinesets += int32(count)
 	}
 	klog.Infof("The nodes allowed: %v and total nodes in cluster after node scale-out %v", maxScaleNodesAllowed, totalNodesAddedByMachinesets)
-	return totalNodesAddedByMachinesets < int32(maxScaleNodesAllowed)
+	return totalNodesAddedByMachinesets <= int32(maxScaleNodesAllowed)
```
Comment (asm582) [inline controllers/appwrapper_controller.go:211]: Master node is also considered, it should be less than for now, unless a filter function is written to not consider master nodes

**Pull request merged**